Assets/PresetsPerFolder/PresetPostProcessor.cs

```diff
@@ -0,0 +1,219 @@
+            if (assetPath.StartsWith("Assets/") && !assetPath.EndsWith(".cs") && !assetPath.EndsWith(".preset"))
+            {
+                var path = Path.GetDirectoryName(assetPath);
+                ApplyPresetsFromFolderRecursively(path);
+            }
+        }
+
+        void ApplyPresetsFromFolderRecursively(string folder)
+        {
+            // Apply everything from the parent folder first so the preset closer to our asset will be applied last.
+            var parentFolder = Path.GetDirectoryName(folder);
+            if (!string.IsNullOrEmpty(parentFolder))
+                ApplyPresetsFromFolderRecursively(parentFolder);
+
+            // Add a dependency to the folder preset custom key
+            // so whenever a Preset is added or removed from this folder our asset is re-imported.
+            context.DependsOnCustomDependency($"PresetPostProcessor_{folder}");
+
+            // Find all Preset assets in this folder. We are using the System.Directory method here instead of the AssetDatabase
+            // because the import may run in a separate process which prevent the AssetDatabase from doing global searching.
+            var presetPaths =
+                Directory.EnumerateFiles(folder, "*.preset", SearchOption.TopDirectoryOnly)
+                    .OrderBy(a => a);
+
+            foreach (var presetPath in presetPaths)
+            {
+                // Load the Preset and try to apply it to the importer.
+                var preset = AssetDatabase.LoadAssetAtPath<Preset>(presetPath);
+
+                // In case we are being imported before the preset, we can't know if we need to depends on it or not.
+                // To be sure, we add a dependency to it anyway so we are being re-imported once it's imported.
+                // If the preset has been loaded correctly, we want to depends on it only if it applies to our asset.
+                if (preset == null || preset.ApplyTo(assetImporter))
+                {
+                    // Using DependsOnArtifact here because Presets are native assets and using DependsOnSourceAsset would not work.
+                    context.DependsOnArtifact(presetPath);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// InitPresetDependencies:
+    /// This method is being called when the project is loaded and find every Preset imported in the project.
+    /// For each folder containing a Preset, we create a CustomDependency from the folder name and a hash being the list of Presets found in it.
+    ///
+    /// OnAssetsModified:
+    /// Whenever a Preset is being added, removed, or moved from a folder, we need to update the CustomDependency for this folder
+    /// so assets that may depend on those presets are re-imported.
+    ///
+    /// TODO: Ideally each CustomDependency should also be dependent on the PresetType,
+    /// so your Texture is not being re-imported by adding a new FBXImporterPreset in a folder.
```

Comment:
Suggestion: "so Textures are not re-imported by adding a new FBXImporterPreset in a folder."